checknoglobals/testdata/src/12/code.go

```diff
@@ -0,0 +1,12 @@
+package code
+
+import "errors"
+
+// myVar is just a bad named global var.
+var myVar = 1 // want "myVar is a global variable"
+
+// ErrNotFound is an error and should be OK.
+var ErrNotFound = errors.New("this is error")
+
+// ErrIsNotErr is an error and should be OK.
+var ErrIsNotErr = 1 // want "ErrIsNotErr is a global variable, should be a const"
```

Comment:
Something I'm noticing here is from the users perspective, knowing little about how we got here, that `myVar` and `ErrIsNotErr` look awfully alike and yet result in different errors.

I think the suggestion "should be a const" probably applies to any basic literal, not just errors. So instead of introducing it specifically as a feature for errors, we introduce it as a feature that applies to any var, as a suggestion. When a var is detected in this fashion, we don't bother checking the other exclusions, except the go:embed one, because any others can be changed to a const.

What do you think?

Reply:
Oh, of course! This is in fact already how it works and why I pass both the variable and it's value when doing the check now. We first search for `*ast.BasicLit` and then we check the name.

The analyzer test package doesn't require the directive to match exactly, only a subset have to match, that's why the tests passed even though it didn't have the full error message.

I updated the test cases for `myVar`, to make this clearer, everything that's assigned to a `*ast.BasicLit` will see "should be a const".